lektor/publisher.py

```diff
@@ -609,6 +609,8 @@ def write_cname(self, path, target_url):
     def publish(self, target_url, credentials=None, **extra):
         if not locate_executable('git'):
             self.fail('git executable not found; cannot deploy.')
+        if target_url.host is None:
+            self.fail('host name missing')
```

Comment:
This seems unrelated to the rest of this pull request. If it's important, can you pull it out into a separate pull request?